internal/daemon/controller/handlers/scopes/scope_service_test.go

```diff
@@ -575,10 +637,337 @@ func TestList(t *testing.T) {
+		),
+	)
+
+	// second page
+	req.ListToken = got.ListToken
+	got, err = s.ListScopes(ctx, req)
+	require.NoError(t, err)
+	require.Len(t, got.GetItems(), 2)
+
+	assert.Empty(t,
+		cmp.Diff(
+			got,
+			&pbs.ListScopesResponse{
+				Items:        wantProjects[2:4],
+				ResponseType: "delta",
+				ListToken:    "",
+				SortBy:       "created_time",
+				SortDir:      "desc",
+				RemovedIds:   nil,
+				EstItemCount: 12,
+			},
+			cmpopts.SortSlices(func(a, b string) bool {
+				return a < b
+			}),
+			cmpopts.SortSlices(func(a, b protocmp.Message) bool {
+				return a.String() < b.String()
+			}),
+			protocmp.Transform(),
+			protocmp.IgnoreFields(&pbs.ListScopesResponse{}, "list_token"),
+		),
+	)
+
+	// remainder of results
+	req.ListToken = got.ListToken
+	req.PageSize = 6
+	got, err = s.ListScopes(ctx, req)
+	require.NoError(t, err)
+	require.Len(t, got.GetItems(), 6)
+
+	assert.Empty(t,
+		cmp.Diff(
+			got,
+			&pbs.ListScopesResponse{
+				Items:        wantProjects[4:],
+				ResponseType: "complete",
+				ListToken:    "",
+				SortBy:       "created_time",
+				SortDir:      "desc",
+				RemovedIds:   nil,
+				EstItemCount: 12,
+			},
+			cmpopts.SortSlices(func(a, b string) bool {
+				return a < b
+			}),
+			cmpopts.SortSlices(func(a, b protocmp.Message) bool {
+				return a.String() < b.String()
+			}),
+			protocmp.Transform(),
+			protocmp.IgnoreFields(&pbs.ListScopesResponse{}, "list_token"),
+		),
+	)
+
+	// create another scope
+	newP, err := iam.NewProject(ctx, oWithProjects.GetPublicId())
+	require.NoError(t, err)
+	p, err := repo.CreateScope(ctx, newP, "")
+	require.NoError(t, err)
+	newScope := &pb.Scope{
+		Id:                          p.GetPublicId(),
+		ScopeId:                     oWithProjects.GetPublicId(),
+		Scope:                       &pb.ScopeInfo{Id: oWithProjects.GetPublicId(), Type: scope.Org.String(), ParentScopeId: scope.Global.String()},
+		CreatedTime:                 p.GetCreateTime().GetTimestamp(),
+		UpdatedTime:                 p.GetUpdateTime().GetTimestamp(),
+		Version:                     1,
+		Type:                        scope.Project.String(),
+		AuthorizedActions:           testAuthorizedActions,
+		AuthorizedCollectionActions: paginationAuthorizedCollectionActions,
+	}
+	// Add to the front of the slice since it's the most recently updated
+	wantProjects = append([]*pb.Scope{newScope}, wantProjects...)
+
+	// delete different scope
+	_, err = repo.DeleteScope(ctx, wantProjects[len(wantProjects)-1].Id)
+	wantProjects = wantProjects[:len(wantProjects)-1]
+	require.NoError(t, err)
+
+	// Run analyze to update postgres estimates
+	_, err = sqlDb.ExecContext(ctx, "analyze")
+	require.NoError(t, err)
+
+	// request updated results
+	// since both creating and deleting scopes will affect the grantsHash
+	// we expect this to error
+	req.ListToken = got.ListToken
+	_, err = s.ListScopes(ctx, req)
+	require.Error(t, err)
+	require.True(t, errors.Match(errors.T(errors.InvalidListToken), err))
```

Comment:
I couldn't get this not to error so I'm leaving it as it was in the previous PR.